test/models-define-type.test.js

```diff
@@ -0,0 +1,67 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: loopback-connector-remote
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+const helper = require('./helper');
+const loopback = require('loopback');
+const sinon = require('sinon');
+
+const relation = require('loopback-datasource-juggler/lib/relation-definition');
+const RelationTypes = relation.RelationTypes;
+
+describe('Models Define Type Tests', function() {
+  let serverApp, clientApp, remoteDs, defineObjectTypeSpy, ChildModel;
+
+  before('create remote datasource', () => {
+    serverApp = helper.createRestAppAndListen();
+    clientApp = loopback({localRegistry: true});
+    remoteDs = helper.createRemoteDataSource(clientApp, serverApp);
+    defineObjectTypeSpy = sinon.spy(remoteDs.connector.remotes,
```

Comment:
This creates a single spy shared by all tests below. As a result, the spy will record multiple invocations, depending on how many tests are being run (e.g. via `mocha -g` or `it.only()`). I am concerned that if something breaks, then unrelated test will start failing because they will see bad state from previous tests.

I am proposing to create a new spy before each test and restore the spied method after each test, or perhaps use your approach from https://github.com/strongloop/loopback-connector-remote/pull/83, where the spy is reset in a beforeEach hook.

Reply:
Fixed